src/app.py

```diff
@@ -963,7 +963,85 @@ def publish_datastage(identifier):
+     --header "Content-Type: application/json" \
+     --header "Accept: application/json" \
+     --header "Authorization: Bearer ${TOKEN}" \
+     --data '{"uuid":["fdacb21245e14fd126bfce05ae191857", "b4ed22922fe9efcc68bb656da04052cb"]}'
+    """
+    if not request.is_json:
+        return Response("json request required", 400)
+    dataset_uuids = request.json.get('uuid')
+    if type(dataset_uuids) is not list or len(dataset_uuids) == 0:
+        return Response("json request must contain a uuid key which is an array of dataset uuids", 400)
+
+    try:
+        auth_helper = AuthHelper.configured_instance(app.config['APP_CLIENT_ID'], app.config['APP_CLIENT_SECRET'])
+        ingest_helper = IngestFileHelper(app.config)
+        user_info = auth_helper.getUserInfoUsingRequest(request, getGroups=True)
+        if user_info is None:
+            return Response("Unable to obtain user information for auth token", 401)
+        if isinstance(user_info, Response):
+            return user_info
+
+        if 'hmgroupids' not in user_info:
+            return Response("User has no valid group information.", 403)
+        if data_admin_group_uuid not in user_info['hmgroupids']:
+            return Response("User must be a member of the HuBMAP Data Admin group.", 403)
+
+        for dataset_uuid in dataset_uuids:
+            q = f"MATCH (e:Dataset {{uuid: '{dataset_uuid}'}}) RETURN " \
+                "e.uuid as uuid, e.entity_type as entitytype, e.status as status, " \
+                "e.data_access_level as data_access_level, e.group_uuid as group_uuid, " \
+                "e.contacts as contacts, e.contributors as contributors, " \
+                "e.ingest_metadata as ingest_metadata"
+            with neo4j_driver_instance.session() as neo_session:
+                rval = neo_session.run(q).data()
+
+                dataset_data_access_level = rval[0]['data_access_level']
+                dataset_group_uuid = rval[0]['group_uuid']
+                dataset_ingest_metadata = rval[0].get('ingest_metadata')
+
+                dataset_ingest_matadata_dict = None
+                if dataset_ingest_metadata is not None:
+                    dataset_ingest_matadata_dict: dict =\
+                        string_helper.convert_str_literal(dataset_ingest_metadata)
+
+                # Save a .json file with the metadata information at the top level directory...
+                if dataset_ingest_matadata_dict is not None:
+                    logger.info(f"ingest_matadata: {dataset_ingest_matadata_dict}")
+                    json_object = json.dumps(dataset_ingest_matadata_dict['metadata'], indent=4)
+                    json_object += '\n'
+                    ds_path = ingest_helper.dataset_directory_absolute_path(dataset_data_access_level,
+                                                                            dataset_group_uuid, dataset_uuid, False)
+                    md_file = os.path.join(ds_path, "metadata.json")
```

Comment:
Should this overwrite the old metadata.json if it already exists?

Reply:
@yuanzhou It does as. you have pointed out. I'm not sure why it should not over write the file. Again I suggest that this is a question for @shirey as I do not have requirements for overwriting yes/no in the card.